src/Logging/HTTPOutputHandler.php

```diff
@@ -167,4 +186,12 @@ protected function write(array $record)
 
         return false === $this->getBubble();
     }
+
+    /**
+     * This method is required and must be protected for unit testing, since we can't mock static or private methods
+     */
+    protected function isCli(): bool
```

Comment:
Get rid of this method and just call Director::is_cli() directly above

Reply:
> This method is required and must be protected for unit testing, since we can't mock static or private methods

Unless you have a way to mock those, this is required.